src/tests/compiler/ast/module/sourceFileTests.ts

```diff
@@ -1405,4 +1405,79 @@ const t = 5;`;
+            const { sourceFile } = getInfoFromText(code, { filePath: "/File.ts" });
+            if (expected === "throw") {
+                expect(() => sourceFile.getLineAndColumnAtPos(pos)).to.throw();
+            }
+            else {
+                expect(sourceFile.getLineAndColumnAtPos(pos)).to.deep.equal(expected);
+            }
+        }
+        it("should return line and column numbers at given position in single line source files", () => {
+            test(``, 0, { line: 1, column: 0 });
+            const code = `interface I { m1(): void }`;
+            test(code, 3, { line: 1, column: 3 });
+            test(code, 0, { line: 1, column: 0 });
+            test(code, code.length, { line: 1, column: code.length });
+        });
+        it("should return line and column numbers at given position in empty first line source files", () => {
+            const code = `\ninterface I { m1(): void }`;
+            test(code, 0, { line: 1, column: 0 });
+            test(code, 1, { line: 2, column: 0 });
+            test(code, code.length, { line: 2, column: code.length - 1 });
+        });
+        it("should return line and column numbers at given position in multiple lines and comments source file", () => {
+            const code = `
+/**
+ * Lorem
+ * @param ipsum - Gives.
+ */
+   // line starting with spaces
+interface I {
+    /**
+     * jsdoc comment 2.
+     * @returns - Lorem ipsum
+     */
+    m1(): void
+    /* block1 */
+}
+  export class 
+    C implements 
+ I { // line starting with spaces
+/* line ending with spaces */ m1(){             
+
+    }
+ }
+// tail
+`;
+            test(code, 0, { line: 1, column: 0 });
+            test(code, 1, { line: 2, column: 0 });
+            test(code, code.indexOf("@param ipsum") - 3, { line: 4, column: 0 });
+            test(code, code.indexOf("@param ipsum") + "@param ipsum - Gives.".length, { line: 4, column: " * @param ipsum - Gives.".length });
+            test(code, code.indexOf("   // line star"), { line: 6, column: 0});
+            test(code, code.indexOf("   // line star") + 2, { line: 6, column: 2});
+            test(code, code.indexOf("/* block1 */"), { line: 13, column: 4});
+            test(code, code.indexOf("C implements") - 1, { line: 16, column: 3});
+            test(code, code.indexOf("/* line ending with spaces */ m1(){      ") + "/* line ending with spaces */ m1(){      ".length,
+                { line: 18, column: "/* line ending with spaces */ m1(){      ".length});
+            test(code, code.indexOf("// tail") + "// tail".length, { line: 22, column: "// tail".length});
+            test(code, code.indexOf("// tail") + "// tail".length + 1, { line: 23, column: 0});
+        });
+
+        it.skip("should throw when position is length + 1", () => {
+            test(`var a = 1`, `var a = 1`.length + 1, "throw");
+            test(``, 1, "throw");
```

Comment:
@dsherret this is the main doubt I have, is createSourceFile adding extra characters since this should throw ? If so, do you know if I can disable that behavior so it throws?

Reply:
I'm not sure. Did you try debugging to check? I can look at it later if not.

Reply:
IMO there is an error in StringUtils methods: 
```ts
    static getLineNumberAtPos(str: string, pos: number) {
        errors.throwIfOutOfRange(pos, [0, str.length + 1], nameof(pos));
```

should be `errors.throwIfOutOfRange(pos, [0, str.length], nameof(pos));`

right?

But then if I change it, the following test fails, and I think the test is wrong: 

```ts
        it("should not throw if providing a pos the length of the string", () => {
            expect(() => StringUtils.getLineNumberAtPos("", 1)).to.not.throw();
        });
```

Will try to fix this and see if tests are green but maybe you have some idea why you implement this this way? is just an error or is it a hack for something to work?

Reply:
@dsherret So if I fix StringUtil methods changing the throwIfOutOfRange and change its tests that test explicitly for this (like previous) everything works as expected and tests are green. But I wonder if fixing this issue won't have consequences for user's code ? Is just that, do you want me to push it right here? or perhaps in a separate PR ?

Reply:
Yeah, that's correct and that test is wrong. Should be `0` and not `1`. It can be in this PR. Thanks! :)

Reply:
Done